erpnext/controllers/tests/test_subcontracting_controller.py

```
@@ -1162,6 +1162,54 @@ def test_phantom_bom_explosion(self):
 
 		self.assertEqual([item.rm_item_code for item in sco.supplied_items], expected)
 
+	def test_co_by_product(self):
+		frappe.set_value("UOM", "Nos", "must_be_whole_number", 0)
+
+		fg_item = make_item("FG Item", properties={"is_stock_item": 1, "is_sub_contracted_item": 1}).name
+		rm_item = make_item("RM Item", properties={"is_stock_item": 1}).name
+		scrap_item = make_item("Scrap Item", properties={"is_stock_item": 1}).name
+		make_bom(
+			item=fg_item, raw_materials=[rm_item], scrap_items=[scrap_item], process_loss_percentage=10
+		).name
```

Ruff (0.15.4)

[warning] Found useless attribute access. Either assign it to a variable or remove it.

(B018)